fetch-in-react/my-app/src/UserDetails.tsx

```diff
@@ -0,0 +1,52 @@
+/* eslint-disable @typescript-eslint/no-unused-vars -- Remove me */
+import { useEffect, useState } from 'react';
+import { UserCard } from './UserCard';
+import type { User } from './Users';
+
+type Props = {
+  userId: number;
+  onCancel: () => void;
+};
+export function UserDetails({ userId, onCancel }: Props) {
+  const [isLoading, setIsLoading] = useState(true);
+  const [error, setError] = useState<unknown>();
+  const [user, setUser] = useState<User>();
+
+  /* your code here (hint: useEffect) */
+  useEffect(() => {
+    async function getUser() {
+      try {
+        const res = await fetch(
+          `https://jsonplaceholder.typicode.com/users/${userId}`
+        );
+        if (!res.ok) {
+          throw new Error(`fetch failed ${res.status}`);
+        }
+        const newuser = await res.json();
+        setUser(newuser);
+        setIsLoading(false);
+      } catch (err) {
+        setError(err);
+      }
+    }
+    getUser();
+    return () => setIsLoading(false);
```

Comment:
No, you don't want that here, you want it in the `catch` (or move it into the `finally` and remove it from the try block). The value you return from useEffect is the clean-up function, which is intended to clean up resources. It doesn't run until your component unmounts. By putting it here, you won't properly report errors to the user.